codice/ddf
# [2.13.x] DDF-4320 Fix CDM ingesting too soon on system start

#### What does this PR do?
- Defines a json file specifying InputTransformer IDs started by default in DDF
- CDM routes will wait to be started until InputTransformer services satisfy those IDs, or until the wait timeout is exceeded.
- Cleans up a few dependencies 

#### Who is reviewing it? 
@emmberk @kcover @brendan-hofmann @pvargas @mdang8 

#### Select relevant component teams:  
@codice/io 

#### Ask 2 committers to review/merge the PR and tag them here.
@clockard
@tbatie 

#### How should this be tested?
Set `log:set TRACE org.codice.ddf.catalog.content.monitor` for detailed logs.
Verify the following scenarios. 
- Start a normal CDM (move/delete/inplace) and verify it begins ingesting
- Start a normal CDM (move/delete/inplace) and verify it begins ingesting on a directory with a large amount of products. Restart the distribution while ingesting and verify it does not start ingesting until all InputTransformers are available (view logs).
- Edit the `ddf-default-input-transformers.json` file to include an ID that doesn't exist, or turn off a bundle providing an InputTransformer already listed. Startup a CDM and verify it times out while waiting and continues ingesting as normal.
- Edit the `ddf-default-input-transformers.json` file to include an ID that doesn't exist, or turn off a bundle providing an InputTransformer already listed. Startup a CDM and then immediately delete it while it is waiting for InputTransformers, verify the CDM does *not* begin ingesting.
- Test WebDav according to https://codice.atlassian.net/browse/DDF-3372

#### Any background context you want to provide?

#### What are the relevant tickets?
[DDF-4320](https://codice.atlassian.net/browse/DDF-4320)

#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
